api/agent/call.go

```diff
@@ -107,17 +116,17 @@ func FromHTTPTriggerRequest(app *models.App, fn *models.Fn, trigger *models.Trig
 			// TODO - this wasn't really the intention here (that annotations would naturally cascade
 			// but seems to be necessary for some runner behaviour
 			Annotations: app.Annotations.MergeChange(fn.Annotations).MergeChange(trigger.Annotations),
-			Headers:     req.Header,
+			Headers:     headers,
 			CreatedAt:   common.DateTime(time.Now()),
-			URL:         reqURL(req),
+			URL:         requestUrl,
 			Method:      req.Method,
 			AppID:       app.ID,
 			AppName:     app.Name,
 			FnID:        fn.ID,
 			TriggerID:   trigger.ID,
 			SyslogURL:   syslogURL,
 		}
-
+		req.Header = headers
```

Comment:
I'd probably do this higher up, as soon as the header rewriting is completed and then just use req.Heade everywhere. Avoid confusion later.